pkg/server/server.go

```diff
@@ -288,43 +293,17 @@ func (s *Server) Run(ctx context.Context) error {
 		return nil
 	})
 
-	//Create Client and Shared
-	var clientConfig clientcmdapi.Config
-	clientConfig.AuthInfos = map[string]*clientcmdapi.AuthInfo{
-		"loopback": {Token: server.LoopbackClientConfig.BearerToken},
-	}
-	clientConfig.Clusters = map[string]*clientcmdapi.Cluster{
-		// cross-cluster is the virtual cluster running by default
-		"cross-cluster": {
-			Server:                   server.LoopbackClientConfig.Host + "/clusters/*",
-			CertificateAuthorityData: server.LoopbackClientConfig.CAData,
-			TLSServerName:            server.LoopbackClientConfig.TLSClientConfig.ServerName,
-		},
-		// admin is the virtual cluster running by default
-		"admin": {
-			Server:                   server.LoopbackClientConfig.Host,
-			CertificateAuthorityData: server.LoopbackClientConfig.CAData,
-			TLSServerName:            server.LoopbackClientConfig.TLSClientConfig.ServerName,
-		},
-		// user is a virtual cluster that is lazily instantiated
-		"user": {
-			Server:                   server.LoopbackClientConfig.Host + "/clusters/user",
-			CertificateAuthorityData: server.LoopbackClientConfig.CAData,
-			TLSServerName:            server.LoopbackClientConfig.TLSClientConfig.ServerName,
-		},
-	}
-	clientConfig.Contexts = map[string]*clientcmdapi.Context{
-		"cross-cluster": {Cluster: "cross-cluster", AuthInfo: "loopback"},
-		"admin":         {Cluster: "admin", AuthInfo: "loopback"},
-		"user":          {Cluster: "user", AuthInfo: "loopback"},
+	externalKubeConfigPath := filepath.Join(s.options.Extra.RootDirectory, s.options.Extra.KubeConfigPath)
+	loopbackClientConfig, externalClientConfig, err := createKubeConfigs(server, externalKubeConfigAdminToken, externalKubeConfigAdminTokenHash, externalKubeConfigPath)
```

Comment:
loopbackKubeConfig, externalKubeConfig

Reply:
Fix in commit 7c3486cb3506a17f776e0a8fc6bcf5376c42d947